### PR TITLE
[4.4] add ability to stop default error handler 

### PR DESCRIPTION
**Pull request by saeideng**
this helps us to stop event with `$event->stopPropagation();`
in https://github.com/cakephp/debug_kit/pull/856

Refs: #16228

### CODE REVIEW

File: src/Error/ErrorTrap.php [hunk 1]
```diff
@@ -126,7 +126,10 @@ public function handleError(
         try {
             // Log first incase rendering or event listeners fail
             $logger->logMessage($error->getLabel(), $error->getMessage());
-            $this->dispatchEvent('Error.handled', ['error' => $error]);
+            $event = $this->dispatchEvent('Error.handled', ['error' => $error]);
```
Comment (othercorey) [inline src/Error/ErrorTrap.php]: If this event prevents rendering, we should call it `beforeRender`.

Comment (markstory) [reply src/Error/ErrorTrap.php]: I am ok with `Error.beforeRender` as well if folks think it is better. It has the benefit of sharing terminology and behavior with controller and view events which is nice.

**Review by markstory — COMMENTED** [review]
Looks good to me. We should add a test to cover the event handler being able to prevent rendering though.

**Comment by markstory** [comment]
I pushed up commits to both this and the DebugKit pull request to:

1. Use `Error.beforeRender` for the event name. 
2. Added tests for stopping `Error.beforeRender`
3. Use `Exception.beforeRender` for consistency with error handling. 
4. In DebugKit, I've switched to `Error.beforeRender` and added event stopping.

**Comment by othercorey** [comment]
Looks good to me!

**Review by saeideng — COMMENTED** [review]
Looks good to me

File: src/Error/ExceptionTrap.php [hunk 2]
```diff
@@ -17,9 +17,13 @@
  * Using the `register()` method you can attach an ExceptionTrap to PHP's default exception handler and register
  * a shutdown handler to handle fatal errors.
  *
- * When exceptions are trapped the `Exception.handled` event is triggered.
+ * When exceptions are trapped the `Exception.beforeRender` event is triggered.
  * Then exceptions are logged (if enabled) and finally 'rendered' using the defined renderer.
  *
+ * Stopping the `Exception.beforeRender` event has no effect, as we always need to render
+ * a response to an exception and custom renderers should be used if you want to replace or
+ * skip rendering an exception..
```
Comment (ADmad) [inline src/Error/ExceptionTrap.php]:
```suggestion
 * skip rendering an exception.
```

**Pull request merged**